### PR TITLE
Added unit test for invalid sprite animation crash

### PR DESCRIPTION
**Pull request by JCash**
## PR checklist

* [ ] Code
	* [ ] Add engine and/or editor unit tests.
	* [ ] New and changed code follows the overall code style of existing code
	* [ ] Add comments where needed
* [ ] Documentation
	* [ ] Make sure that API documentation is updated in code comments
	* [ ] Make sure that manuals are updated (in github.com/defold/doc)
* [ ] Prepare pull request and affected issue for automatic release notes generator
	* [ ] Pull request - Write a message that explains what this pull request does. What was the problem? How was it solved? What are the changes to APIs or the new APIs introduced? This message will be used in the generated release notes. Make sure it is well written and understandable for a user of Defold.
	* [ ] Pull request - Write a pull request title that in a sentence summarises what the pull request does. Do not include "Issue-1234 ..." in the title. This text will be used in the generated release notes.
	* [ ] Pull request - Link the pull request to the issue(s) it is closing. Use on of the [approved closing keywords](https://docs.github.com/en/issues/tracking-your-work-with-issues/linking-a-pull-request-to-an-issue).
	* [ ] Affected issue - Assign the issue to a project. Do not assign the pull request to a project if there is an issue which the pull request closes.
	* [ ] Affected issue - Assign the "breaking change" label to the issue if introducing a breaking change.
	* [ ] Affected issue - Assign the "skip release notes" is the issue should not be included in the generated release notes.

----------

Example of a well written PR description:

1. Start with the user facing changes. This will end up in the release notes.
1. Add one of the GitHub approved closing keywords
1. Optionally also add the technical changes made. This is information that might help the reviewer. It will not show up in the release notes. Technical changes are identified by a line starting with one of these:
   1. `### Technical changes` 
   1. `Technical changes:`
   2. `Technical notes:`

```
There was a anomaly in the carbon chroniton propeller, introduced in version 8.10.2. This fix will make sure to reset the phaser collector on application startup.

Fixes #1234

### Technical changes
* Pay special attention to line 23 of phaser_collector.clj as it contains some interesting optimizations
* The propeller code was not taking into account a negative phase.
```

### CODE REVIEW

**Review by AGulev — DISMISSED** [review]
❤️

File: engine/gamesys/src/gamesys/test/sprite/sprite_flipbook_anim.script [hunk 1]
```diff
@@ -18,24 +18,32 @@ num_messages = 0
 
 local function callback(self, message_id, message, sender)
     num_finished = num_finished + 1
-    tests_done = num_finished == 2
+
+    self.waiting_to_render_invalid_texture = 1
+    sprite.play_flipbook("#sprite", "does_not_exist") -- see to it that we handle an invalid animation
```
Comment (JCash) [inline engine/gamesys/src/gamesys/test/sprite/sprite_flipbook_anim.script:23]: Invoke the bad behavior

File: engine/gamesys/src/gamesys/test/test_gamesys.cpp [hunk 2]
```diff
@@ -954,17 +954,7 @@ TEST_F(SpriteTest, FlipbookAnim)
...
-    {
-        ASSERT_TRUE(dmGameObject::Update(m_Collection, &m_UpdateContext));
-        ASSERT_TRUE(dmGameObject::PostUpdate(m_Collection));
-
-        // check if tests are done
-        lua_getglobal(L, "tests_done");
-        tests_done = lua_toboolean(L, -1);
-        lua_pop(L, 1);
-    }
+    WaitForTestsDone(10000, true, 0);
```
Comment (JCash) [inline engine/gamesys/src/gamesys/test/test_gamesys.cpp:957]: To make it easier, I've created a helper function to wait for the result, and also correctly update+render the systems.

Previously, we only updated the systems, but never called into the render parts that contained the bug.

File: engine/gamesys/src/gamesys/test/test_gamesys.cpp [hunk 3]
```diff
@@ -993,7 +983,7 @@ TEST_F(SpriteTest, FrameCount)
     dmGameObject::HInstance go = Spawn(m_Factory, m_Collection, "/sprite/frame_count/sprite_frame_count.goc", dmHashString64("/go"), 0, 0, Point3(0, 0, 0), Quat(0, 0, 0, 1), Vector3(1, 1, 1));
     ASSERT_NE((void*)0, go);
 
-    WaitForTestsDone(100, 0);
+    WaitForTestsDone(100, false, 0);
```
Comment (JCash) [inline engine/gamesys/src/gamesys/test/test_gamesys.cpp:986]: Some tests are not written to work with rendering (yet)

File: engine/gamesys/src/gamesys/test/test_gamesys.h [hunk 4]
```diff
@@ -433,6 +434,7 @@ void GamesysTest<T>::SetUp()
     dmResource::RegisterTypes(m_Factory, &m_Contexts);
 
     dmGraphics::InstallAdapter();
+    dmGraphics::ResetDrawCount(); // for the unit test
```
Comment (JCash) [inline engine/gamesys/src/gamesys/test/test_gamesys.h:437]: Since more tests use the rendering, we need to be able to reset the unit test flag.

**Pull request merged**